sticker-utils/src/bin/sticker-train.rs

```diff
@@ -118,113 +176,59 @@ fn train_model(
     Ok(())
 }
 
-fn run_epoch(
+fn run_epoch<E>(
+    config: &Config,
+    encoder: &mut CategoricalEncoder<E, E::Encoding>,
+    vectorizer: &SentVectorizer,
     trainer: &TaggerTrainer,
-    tensors: &CollectedTensors,
+    file: &mut File,
```

Comment:
Reference to the file is kept instead of re-opening the file after each epoch, so external changes to the file don't mess up the training?

Reply:
Depends on the kind of changes (does it change the file in-place or not). But at least on unixes, you can remove or move the file and the file descriptor will continue to be valid as long as it's open.